metagpt/actions/write_code.py

```diff
@@ -101,22 +112,39 @@ async def run(self, *args, **kwargs) -> CodingContext:
 
         if bug_feedback:
             code_context = coding_context.code_doc.content
+        elif code_plan_and_change:
+            code_context = await self.get_codes(
+                coding_context.task_doc, exclude=self.i_context.filename, project_repo=self.repo, mode="incremental"
+            )
         else:
             code_context = await self.get_codes(
                 coding_context.task_doc,
                 exclude=self.i_context.filename,
                 project_repo=self.repo.with_src_path(self.context.src_workspace),
             )
 
-        prompt = PROMPT_TEMPLATE.format(
-            design=coding_context.design_doc.content if coding_context.design_doc else "",
-            tasks=coding_context.task_doc.content if coding_context.task_doc else "",
-            code=code_context,
-            logs=logs,
-            feedback=bug_feedback.content if bug_feedback else "",
-            filename=self.i_context.filename,
-            summary_log=summary_doc.content if summary_doc else "",
-        )
+        if code_plan_and_change:
+            prompt = REFINED_TEMPLATE.format(
+                user_requirement=requirement_doc.content if requirement_doc else "",
+                code_plan_and_change=code_plan_and_change,
+                design=coding_context.design_doc.content if coding_context.design_doc else "",
+                task=coding_context.task_doc.content if coding_context.task_doc else "",
+                code=code_context,
+                logs=logs,
+                feedback=bug_feedback.content if bug_feedback else "",
+                filename=self.i_context.filename,
+                summary_log=summary_doc.content if summary_doc else "",
+            )
+        else:
+            prompt = PROMPT_TEMPLATE.format(
+                design=coding_context.design_doc.content if coding_context.design_doc else "",
+                task=coding_context.task_doc.content if coding_context.task_doc else "",
```

Comment:
`tasks`? The place holder named `task` doesn't exists.

Reply:
Tasks have been modified to task in dev
![image](https://github.com/geekan/MetaGPT/assets/52203545/73ad9e03-90c4-4a00-ad1a-7c22d96722c4)

Reply:
See the definition of `PROMPT_TEMPLATE`, it's named `tasks`
```
PROMPT_TEMPLATE = """
......
## Tasks
{tasks}
......
```